tests/component/test_watchdog.py

```diff
@@ -0,0 +1,81 @@
+from nidaqmx import Task
+from nidaqmx.constants import WatchdogAOExpirState, WatchdogCOExpirState
+from nidaqmx.system import Device
+from nidaqmx.system.watchdog import AOExpirationState, COExpirationState
+
+
+def test___call_cfg_watchdog_ao_expir_states___no_error(
+    generate_watchdog_task: Task, sim_9189_device: Device, sim_9263_device: Device
+):
+    watchdog_task = generate_watchdog_task(f"{sim_9189_device.name}", timeout=0.8)
+    expir_states = [
+        AOExpirationState(
+            physical_channel=sim_9263_device.ao_physical_chans[0].name,
+            expiration_state=0.0,
+            output_type=WatchdogAOExpirState.VOLTAGE,
+        ),
+        AOExpirationState(
+            physical_channel=sim_9263_device.ao_physical_chans[1].name,
+            expiration_state=0.0,
+            output_type=WatchdogAOExpirState.VOLTAGE,
+        ),
+    ]
+    watchdog_task.cfg_watchdog_ao_expir_states(expir_states)
+    watchdog_task.start()
+
+    assert not watchdog_task.expired
+    assert watchdog_task.timeout == 0.8
+    assert (
+        watchdog_task.expiration_states[sim_9263_device.ao_physical_chans[1].name].ao_state == 0.0
+    )
+    assert (
+        watchdog_task.expiration_states[sim_9263_device.ao_physical_chans[1].name].ao_output_type
+        == WatchdogAOExpirState.VOLTAGE
+    )
```

Comment:
The assert section should correspond to the "then" clause. None of these assertions are related to "no_error".

assert timeout is validating a parameter passed to the constructor, which is not what you are testing (cfg_watchdog_ao_expir_states).

assert ao_state  and ao_output_type are reading back properties set by cfg_watchdog_ao_expir_states, which is a good thing to test. Rename the "then" clause to "_sets_expir_state_properties".

I'm not sure about asserting expired. Does cfg_watchdog_ao_expir_states clear the expired property?

Reply:
I was just testing that I could call the function without an error. With a simulated device expired never becomes true.

How else might I test `DAQmxControlWatchdogTask()`? clear_exporation and reset_timer are the only Python functions that call it.

Other tests, I changed the then clause as suggested.

Reply:
> How else might I test DAQmxControlWatchdogTask()?

In a test function that has a "then" clause of "control_watchdog_task".

`test___call_cfg_watchdog_ao_expir_states___no_error` should test DAQmxCfgWatchdogAOExpirStates and nothing else.

Reply:
The test cases for WatchdogTask.clear_expiration and WatchdogTask.reset_timer will test DAQmxControlWatchdogTask.

Reply:
To really test the functionality of DAQmxControlWatchdogTask, you need to use a real device. If you do this, make it a DIO test using the 6363, not an AO test. (I think only cDAQ supports AO watchdogs.)

Reply:
There is a legacy test that calls `reset_timer()` with real HW. Between that and calling `clear_expiration()` on a simulated device and not getting an error sufficient testing for `DAQmxControlWatchdogTask()`

Reply:
ok